### DISCUSSION TITLE
Is there a way to rotate db credentials in the client?

### DISCUSSION THREAD

**Discussion by dblackhall-tyro** [prisma/prisma#6866] [Q&A] [answered]
Hi, I am taking a look at Prisma for the first time and I am looking for a way to rotate the credentials used by the `PrismaClient`. I need to do this because we are using vault to generate short lived db credentials.
We currently use sequelize and achieve this using a `beforeConnect` hook which allows the mutation of the username and password the sequelize client uses to establish the db connection.
I have been searching the docs and experimenting but I haven't found a solution yet. Is this something that is possible with Prisma?

**Comment by 2color** [comment, marked as answer]
Hey @dblackhall-tyro,

Normally, Prisma Client connects to the database using the `DATABASE_URL` environment variable and supports loading that from the `.env` file.

To answer your question, you can programmatically set the connection URL that contains the DB credentials when instantiating Prisma Client:

```ts
const prisma = new PrismaClient({
  datasources: {
    db: {
      url: '....'
    }
  }
})
```

I hope that helps. Let me know if you have any more questions.

**Reply by pedro757** [reply]
This will only work for the client, the CLI won't work, is there a programmatic way to set DATABASE_URL for the CLI too?

When I try "prisma db push" it says I have to set the enviroment variable

@2color

**Reply by janpio** [reply]
https://unix.stackexchange.com/questions/56444/how-do-i-set-an-environment-variable-on-the-command-line-and-have-it-appear-in-c

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
